var/spack/repos/spack_repo/builtin/packages/namd/package.py

```diff
@@ -220,9 +224,19 @@ def _edit_arch_generic(self, spec, prefix):
                             "NAMD_ARCH = {0}".format(self.arch),
                             "CHARMARCH = {0}".format(self.spec["charmpp"].charmarch),
                             "CXX = {0.cxx} {0.cxx11_flag}".format(self.compiler),
-                            "CXXOPTS = {0}".format(optim_opts),
+                            "CXXOPTS = {0} {1} {2} {3}".format(
+                                optim_opts,
+                                " ".join(spec.compiler_flags["cxxflags"]),
+                                " ".join(spec.compiler_flags["cppflags"]),
```

Comment:
cppflags (preprocessor flags) should go before optim_opts both here and for COPTS below.